### ISSUE TITLE
Export survey results to PDF

### ISSUE THREAD

**Issue by kristeine** [webkom/lego-webapp#1305]
I'd love to be able to save survey results as PDF, which is not currently a feature. I've tried "printing" in Chrome with "Save to PDF" as the destination, but the charts break and lose their legends 😢

**Comment by SmithPeder** [comment]
Yeah that would be great. I looked into it a bit, seems like bit of a pain to implement. Might need a few dependencies, but I found a pritty nice flow where you could select a part of the DOM then,
1. Convert the DOM into svg: `html2canvas`
2. Convert the svg into png: `vanilla js`
3. Convert the png into pdf: `jsPDF`

It looks/seems a bit messy, but this would render the whole PDF client side, so that would be pretty cool.

**Comment by odinuge** [comment]
Totally agree!! :smile: 

Including @SmithPeder's solution, I can se two temporary alternatives:

### Puppeteer
The best solution is to use `puppeteer` to generate pdf (or png). There are still a few glitches with the legends, but it looks way better than the normal pdf-printing. The screenshot in `png` looks perfect, but it is still a png.
```js
const puppeteer = require('puppeteer');
const url = "https://abakus.no";
(async () => {
  const browser = await puppeteer.launch();
  const page = await browser.newPage();
  await page.setViewport({ width: 1920, height: 1200 }); // Can also scale here to get better ppi
  await page.goto(url);
  await page.screenshot({ fullPage: true, path: 'page.png' });
  await page.pdf({
    path: 'page.pdf',
    format: 'A4',
    scale: 0.4,
    printBackground: true
  });

  browser.close();
})();
```

### Full-page screenshot

It is also possible to take full screen screenshots using chrome. Still png tho.
https://developers.google.com/web/updates/2017/04/devtools-release-notes#screenshots

**Comment by SmithPeder** [comment]
We could always implement this using the `New'st trick in the book`, by creating the puppeteer implementation as a function :100: Then we can use @JonasBak latex implementation to generate a PDF out of the PNG images :+1: 
> DOM->PDF-as-a-service

**Comment by JonasBak** [comment]
I think a good option, if possible, would be to use puppeteer or something to "export" just tables and graphs to images or pdfs, and then use a 'kvittering'-like function to put all images together to a pdf. This way it is easy to create pdf templates and stuff like that.
